app/models/machine_process.rb

```diff
@@ -90,21 +90,21 @@ class ResurrectionExpiredError < StandardError; end
     end
   end
 
-  scope :for_product, -> product { joins(license: :product).where(products: product) }
-  scope :for_license, -> license { joins(:license).where(licenses: license) }
-  scope :for_machine, -> machine { where(machine: machine) }
-  scope :for_user,    -> id { joins(:license).where(licenses: { user_id: id }) }
+  scope :for_product, -> product { joins(license: :product).where(product: { id: product }) }
+  scope :for_license, -> license { joins(:license).where(license: { id: license }) }
+  scope :for_machine, -> machine { where(machine:) }
+  scope :for_user,    -> id { joins(:license).where(license: { user_id: id }) }
```

Comment:
Drop these modifications. They're unrelated to the upgrade and are going to cause conflicts with #774.

Reply:
Apparently they're not unrelated...

Reply:
What is Rails doing here that causes `where(foo:)` to error while `where(foo: { id: })` doesn't? I'd expect these to be synonymous.

Reply:
It's because in this case, these model scopes can be called from controller scopes, and the controller provides IDs, not model records, so we need to query by `id:` to support both ID and record queries.

Reply:
https://github.com/rails/rails/pull/50829